File: Python/remove_out_par.py
s = input()
ctr = 0
for i in s:
    if i == '(':
        if ctr != 0:
            print(i, end='')
        ctr += 1
    elif i == ')':
        ctr -= 1
        if ctr != 0:
            print(i, end='')

File: C++/remove_out_par.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    cin >> s;
    int ctr = 0;
    for(char c : s){
        if(c=='('){
            if(ctr!=0){
                cout << c;
            }
            ctr++;
        }
        else{
            ctr--;
            if(ctr!=0){
                cout << c;
            }
        }
    }
    cout << endl;
	return 0;
}
